### PR TITLE
Delete & Updated and Select filters

### PR DESCRIPTION
**Pull request by AbbasSarwar**
### Features
* ROLLBACK
* COMMIT
* UPDATING
* REMOVING
* SELECTING BY GROUPS
* AGGREGATE

Screenshots
avgunder2000
avgweight
counts
deletedall
deletes
digimon pokemon
escapedmost
escapednever
history
minANDmax
multipliedby-1
showingagain

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -1,4 +1,12 @@
 INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg ) VALUES (1, 'Agumon', '2020-02-03', 0, B'1', 10.23);
 INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg ) VALUES (2, 'Gabumon', '2018-11-03', 2, B'1', 8);
 INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg ) VALUES (3, 'Pikachu', '2021-01-07', 1, B'0', 15.04);
-INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg ) VALUES (4, 'Devimon', '2017-05-23', 5, B'1', 11);
\ No newline at end of file
+INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg ) VALUES (4, 'Devimon', '2017-05-23', 5, B'1', 11);
+
+INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg ) VALUES (5, 'Charmander', '2020-02-08', 0, B'0', 11);
```
Comment (brytebee) [inline data.sql]: The weight of this animal is `-11` negative. Kindly update.

File: queries.sql [hunk 2]
```diff
@@ -5,4 +5,65 @@ SELECT date_of_birth from animals WHERE name IN ('Agumon', 'Pikachu');
 SELECT name, escape_attempts from animals WHERE weight_kg > 10.5;
 SELECT * from animals WHERE neutered = '1';
 SELECT * from animals WHERE name != 'Gabumon';
-SELECT * from animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+SELECT * from animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
```
Comment (brytebee) [inline queries.sql:9]: You have missed a transaction where you are required to set species to `unspecified`. Take a look.

image

File: queries.sql [hunk 3]
```diff
@@ -5,4 +5,65 @@ SELECT date_of_birth from animals WHERE name IN ('Agumon', 'Pikachu');
...
+begin
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species is null;
+
```
Comment (brytebee) [inline queries.sql:27]: Write a command to verify if the changes in the transaction were made before and after commit. See a snippet

```sql
SELECT species from animals;
```

File: queries.sql [hunk 4]
```diff
@@ -5,4 +5,65 @@ SELECT date_of_birth from animals WHERE name IN ('Agumon', 'Pikachu');
...
+commit
+
+
+begin
+
+DELETE from animals;
+
```
Comment (brytebee) [inline queries.sql:37]: Verify here to before and after `ROLLBACK`.

File: queries.sql [hunk 5]
```diff
@@ -5,4 +5,65 @@ SELECT date_of_birth from animals WHERE name IN ('Agumon', 'Pikachu');
...
+// QUERIES SELECT AGAIN
+
+SELECT COUNT(*) from animals;
+
+SELECT COUNT(*) from animals
+WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg) from animals
+
+SELECT MAX(name) from animals
```
Comment (brytebee) [inline queries.sql]: I believe this should the max escape attempt and not max name.
```sql
SELECT MAX(escape_attempts) from animals
```

File: queries.sql [hunk 6]
```diff
@@ -5,4 +5,65 @@ SELECT date_of_birth from animals WHERE name IN ('Agumon', 'Pikachu');
...
+commit;
+
+
+// QUERIES SELECT AGAIN
+
+SELECT COUNT(*) from animals;
+
+SELECT COUNT(*) from animals
+WHERE escape_attempts = 0;
+
```
Comment (brytebee) [inline queries.sql:69]: You have missed a query to get the average escape_attempts.

```sql
SELECT neutered, AVG(escape_attempts) FROM animals
   GROUP BY neutered;
```

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @AbbasSarwar👋,

##

### Highlights💡
- Great SQL codes to populate DB.✔️
- Good practice of SQL `Transactions` and `rollbacks`.✔️
- Good commit message.✔️ 
- Good PR Summary.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (AbbasSarwar) [reply data.sql]: Changed

File: queries.sql (hunk 2 above)
Comment (AbbasSarwar) [reply queries.sql:9]: unspecified
ROLLEDBACK

File: queries.sql (hunk 3 above)
Comment (AbbasSarwar) [reply queries.sql:27]: added

File: queries.sql (hunk 5 above)
Comment (AbbasSarwar) [reply queries.sql]: maxescape

File: queries.sql (hunk 4 above)
Comment (AbbasSarwar) [reply queries.sql:37]: deletedall
againrolledback

File: queries.sql (hunk 6 above)
Comment (AbbasSarwar) [reply queries.sql:69]: maxs

File: queries.sql [hunk 7]
```diff
@@ -5,4 +5,82 @@ SELECT date_of_birth from animals WHERE name IN ('Agumon', 'Pikachu');
...
-SELECT * from animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+SELECT * from animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+begin
+
+UPDATE animals
+SET species = 'unspecified'
+ROLLBACK
+
```
Comment (TadesseDev) [inline queries.sql]: - [ ] Good job making the changes as requested here to `UPDATE` the animals table by setting the species column to unspecified, but as mentioned earlier for this step you are explicitly expected to verify that change was made and rollbacked, so make sure that you included the `SELECT species from animals;` before and after the `ROLLBACK` statement.

File: queries.sql [hunk 8]
```diff
@@ -5,4 +5,82 @@ SELECT date_of_birth from animals WHERE name IN ('Agumon', 'Pikachu');
...
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species is null;
+
+commit
+
+SELECT * from animals
```
Comment (TadesseDev) [inline queries.sql]: - [ ] Good job adding the select statement after the `commit` statement to verify the change. but as mentioned earlier we need a command to verify if the changes in the transaction were made **before** and after commit. for that alongside the `SELECT * from animals` after `commit` please add one before the `commit` statement.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @AbbasSarwar 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

199733072-bdacca19-8df9-43a7-be93-3aff2f1c15c8

 ## Highlights
 - [x] Good job making most of the changes requested by the previous reviewer. 🙏


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rustam-tolipov — APPROVED** [review]
Hi @AbbasSarwar, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯
3. Well-implemented code based on request 🔢 
4. Great readme 👍 

## Optional suggestions
No optional suggestions :)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**